### PR TITLE
Add Unit Test for OfflineAreaViewerViewModel

### LINKED ISSUE #2596: [Code health] Add Unit Test for OfflineAreaViewerViewModel
Add Unit Test for OfflineAreaViewerViewModel

### PR DESCRIPTION
**Pull request by anandwana001**
Fixes #2596 

@shobhitagarwal1612 @sufyanAbbasi  PTAL?

Adding SS as I am unable to find my branch in codecov
Screenshot 2024-07-31 at 1 37 34 PM

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt [hunk 1]
```diff
@@ -34,13 +34,12 @@ import kotlinx.coroutines.launch
 @AndroidEntryPoint
 class OfflineAreaViewerFragment @Inject constructor() : AbstractMapContainerFragment() {
 
-  private lateinit var viewModel: OfflineAreaViewerViewModel
+  @Inject lateinit var viewModel: OfflineAreaViewerViewModel
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt:37]: Why is this change required?

Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt:37]: When we can inject the viewModel, why to depend on creation?
Sort of a migration towards good practice

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt:37]: Won't this create a new instance of viewModel whenever the fragment is destroyed and recreated? The existing code allows the same view model to be reused across lifecycle changes.

Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt:37]: > fragment is destroyed and recreated

Are we just talking about orientation change or recreation as a whole in a single app session?

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt:37]: I'd recommend not doing this change in this PR since this is out of scope. Let's do it separately while ensuring there are no side-effects.

Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt:37]: ok,

Just tried both ways, new instance will always get created

Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragment.kt]: reverted this part

File: ground/src/test/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import kotlinx.coroutines.test.advanceUntilIdle
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class OfflineAreaViewerViewModelTest : BaseHiltTest() {
```
[the remark this answers is not in the document]
Comment (anandwana001) [reply ground/src/test/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModelTest.kt:32]: The last I talked to @sufyanAbbasi was to test more ViewModel as they contain independent logic and once all is done, we move to the fragment side.

So, in brief the plan is to have independent test which IMO is a good approach. 

WDYT?

[the remark this answers is not in the document]
Comment (anandwana001) [reply ground/src/test/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModelTest.kt:32]: Also, these are not UI test as we are testing the ViewModel variables and functions, values are correct or not.

It could be UI test if we are confirming that the values are visible on the screen by opening a fragment.

**Comment by anandwana001** [comment]
Can we check what is failing?

I don't have access to gcb

**Comment by shobhitagarwal1612** [comment]
Remote build errors:

```

* What went wrong:
Execution failed for task ':ground:kaptGenerateStubsDevDebugUnitTestKotlin'.
> Could not resolve all files for configuration ':ground:detachedConfiguration67'.
   > A build operation failed.
         Could not read workspace metadata from /workspace/.gradle/caches/transforms-4/a3504cfd604fe13b8756dda1d0a49733/metadata.bin
      > Could not read workspace metadata from /workspace/.gradle/caches/transforms-4/a3504cfd604fe13b8756dda1d0a49733/metadata.bin

```

**Review by shobhitagarwal1612 — APPROVED** [review]
